src/cli/cli.cpp

```diff
@@ -4891,32 +4884,43 @@ void Interpreter::ProcessLine(char *aBuf)
     VerifyOrExit(argsLength >= 1);
 
 #if OPENTHREAD_CONFIG_DIAG_ENABLE
-    VerifyOrExit((!otDiagIsEnabled(mInstance) || (strcmp(args[0], "diag") == 0)),
+    VerifyOrExit((!otDiagIsEnabled(mInstance) || (args[0] == "diag")),
                  OutputLine("under diagnostics mode, execute 'diag stop' before running any other commands."));
 #endif
 
-    command = Utils::LookupTable::Find(args[0], sCommands);
+    command = Utils::LookupTable::Find(args[0].GetCString(), sCommands);
 
     if (command != nullptr)
     {
         OutputResult((this->*command->mHandler)(argsLength - 1, &args[1]));
         ExitNow();
     }
 
-    // Check user defined commands if built-in command has not been found
+    SuccessOrExit(ProcessUserCommands(argsLength, args));
+    OutputResult(OT_ERROR_INVALID_COMMAND);
```

Comment:
@abtink In the case where an application has UserCommands registered, it looks like this line would be executed even after a matching command has been successfully executed. This looks like a bug to me as we were able to reproduce it in one of our sample apps. Could you double check?

Reply:
Thanks @lmnotran. Yes. That looks incorrect. Will submit a fix for this soon.

Reply:
Just submitted https://github.com/openthread/openthread/pull/6562.